common/src/main/java/me/shedaniel/architectury/registry/trade/TradeRegistry.java

```diff
@@ -59,4 +144,38 @@ public static void registerTradeForWanderingTrader(boolean rare, VillagerTrades.
         throw new AssertionError();
     }
     
+    /**
+     * Override the max possible offers the wandering trader can have. This does not affect the rare trade.
+     * @param maxOffers  Max possible offers a villager can have.
+     */
+    public static void overrideWanderingTraderMaxOffers(int maxOffers) {
+        if (maxOffers < 0) {
+            throw new IllegalArgumentException("Wandering trader's max offers has to be at least 0!");
+        }
+        
+        WANDERING_TRADER_MAX_OFFER_OVERRIDE = maxOffers;
+    }
+    
+    /**
+     * @return Max offers for the wandering trader. Returning null means no override exists
+     */
+    public static Integer getWanderingTraderMaxOffers() {
+        return WANDERING_TRADER_MAX_OFFER_OVERRIDE;
+    }
+    
+    public static boolean invokeVillagerOfferRemoving(VillagerTradeOfferContext ctx) {
```

Comment:
These should not be publicly accessible to users imo, or at the very least not be included in the same class as the user-facing API. Consider moving this (as well as perhaps the maps themselves) to another class marked as internal.

Reply:
With this. I would move everything into a `trading` package and just leave `MerchantOfferAccess` and the `TradeRegistry` left in the current package? And then move the invoke stuff into his own class. Something against this?

Reply:
I'd rather have the package be called `internal` or `impl` as we **almost definitely** don't want users to call this themselves, so highlighting this with a different package and `@ApiStatus.Internal` might help with that